src/Aspire.Hosting/Dashboard/DashboardLifecycleHook.cs

```diff
@@ -154,4 +166,104 @@ private void ConfigureAspireDashboardResource(IResource dashboardResource)
             }
         }));
     }
+
+    private async Task SubscribeDashboardLogsAsync()
+    {
+        string? dashboardResourceId = null;
+
+        try
+        {
+            using var cts = new CancellationTokenSource(TimeSpan.FromMinutes(1));
+            await foreach (var @event in resourceNotificationService.WatchAsync(cts.Token))
+            {
+                // We want to capture the replica of the dashboard resource. This will allow us to get the logs for the dashboard.
+                // When the resource id is not the same as the resource name, it's the replica.
+                if (@event.Resource.Name == KnownResourceNames.AspireDashboard && @event.ResourceId != @event.Resource.Name)
+                {
+                    dashboardResourceId = @event.ResourceId;
+                    break;
+                }
+            }
+        }
+        catch (OperationCanceledException)
+        {
+            distributedApplicationLogger.LogError("Failed to get the dashboard logs.");
+            return;
+        }
+
+        if (dashboardResourceId is null)
+        {
+            return;
+        }
+
+        var defaultDashboardLogger = loggerFactory.CreateLogger("Aspire.Hosting.Dashboard");
+
+        var loggerCache = new Dictionary<string, ILogger>();
+
+        try
+        {
+            // We turned on the JSON formatter for the logger, so we can log the log lines as JSON.
+            await foreach (var batch in resourceLoggerService.WatchAsync(dashboardResourceId).ConfigureAwait(false))
```

Comment:
This probably stops when things are shutting down, but I think it would be good to have a CTS that is passed here and canceled when the lifecycle hook is stopping.

Reply:
Done